src/optimizer_interface.jl

```diff
@@ -27,12 +27,33 @@ end
 
 
 """
-    function optimize!(model::Model,
-                       optimizer_factory::Union{Nothing, OptimizerFactory}=nothing;
-                       ignore_optimize_hook=(model.optimize_hook === nothing))
+    optimize!(model::Model,
+              optimizer_factory::Union{Nothing, OptimizerFactory}=nothing;
+              ignore_optimize_hook=(model.optimize_hook === nothing))
 
-Optimize the model. If `optimizer_factory` is not `nothing`, it first set the
-optimizer to a new one created using the optimizer factory.
+Optimize the model. If `optimizer_factory` is not `nothing`, it first sets the
+optimizer to a new one created using the optimizer factory. The factory can be
+created using the [`with_optimizer`](@ref) function.
+
+## Examples
+
+The optimizer factory can either be given in the [`Model`](@ref) constructor
+as follows:
+```julia
+model = Model(with_optimizer(GLPK.Optimizer))
+# ...fill model with variables, constraints and objectives...
+# Solve the model with GLPK
+JuMP.optimize!(model)
+```
+or in the `optimize!` call as follows:
+```julia
+model = Model()
+# ...fill model with variables, constraints and objectives...
+# Solve the model with GLPK
+JuMP.optimize!(model, with_optimizer(GLPK.Optimizer))
+# Solve the model with Clp
```

Comment:
Why is the same model solved twice?

Reply:
You are right, it won't work, it will throw `"An optimizer factory cannot both be provided in the `Model` constructor and at the `optimize` call."` at the second call